dotcom-rendering/src/web/components/Picture.tsx

```diff
@@ -204,6 +201,41 @@ const portraitImmersiveSource = (
 	/>
 );
 
+// Create sourcesets for landscape immersive
+const landscapeImmersiveMainMediaSource = (
+	sources: SrcSetItem[],
+	resolution: ResolutionType,
+) => {
+	const sortedSources = sources.slice().sort((a, b) => b.width - a.width);
+
+	const getMedia = (sourceWidth: number, isLast: boolean): string => {
+		const width = resolution === 'hdpi' ? sourceWidth / 2 : sourceWidth;
+		const breakpoint = isLast ? 0 : width;
+
+		return resolution === 'hdpi'
+			? `(min-width: ${breakpoint}px) and (-webkit-min-device-pixel-ratio: 1.25), (min-width: ${breakpoint}px) and (min-resolution: 120dpi)`
+			: `(min-width: ${breakpoint}px)`;
+	};
+
+	return (
+		<>
+			{/* For immersive main media, images will always take up 100% of the screen width,
+				therefor it is more beneficial to loop through all the image sources, instead of the breakpoints.
+				This means that we are not limited to sources within the boundaries of the breakpoints, and can provide
+				more appropriately sized images where applicable. */}
+			{sortedSources.map((sourceSet, index) => (
+				<source
+					srcSet={`${sourceSet.src} ${sourceSet.width}w`}
```

Comment:
Is that a stray w in there after `${sourceSet.width}`? Apologies if it was intentional! 😄

Reply:
Yes sorry this is intentional ~ it's part of the syntax of the `srcSet` attribute!